### ISSUE TITLE
Cache restore failure

### ISSUE THREAD

**Issue by develop7** [nix-community/cache-nix-action#16]
Repro:

Invoke `cache-nix-action` following the example from it's manual:

```yaml
    - name: Restore and cache Nix store
      uses: nix-community/cache-nix-action@v4
      with:
        key: cache-${{ runner.os }}-${{ hashFiles('nix/**/*.nix') }}
        restore-keys: |
          cache-${{ runner.os }}-
        gc-linux: true
        gc-max-store-size-linux: 1000000000
        gc-macos: true
        gc-max-store-size-macos: 1000000000
        purge: true
        purge-keys: cache-${{ runner.os }}-
        purge-created: true
        purge-created-max-age: 42
```

https://github.com/develop7/postgrest/blob/abfeda7512395ee5cf091378fc75f6d223e6833a/.github/actions/setup-nix/action.yaml#L17-L30

Expected: cache is unpacking successfully

Instead: the action output log is polluted with the messages like ones below indicating (partial?) failure to unpack the cache

```
/usr/bin/tar: ../../../../../nix/store/br1p5pan2pgmgrm81kj43qawd9b9nns1-libunistring-1.1/lib/libunistring.so.5.0.0: Cannot open: File exists
```

Link: https://github.com/PostgREST/postgrest/actions/runs/6709349886/job/18232202233?pr=2992#step:3:76

**Comment by deemp** [comment]
Hi! See the [Limitations](https://github.com/nix-community/cache-nix-action#limitations) section.
![image](https://github.com/nix-community/cache-nix-action/assets/48378098/67275b51-999a-4e7e-b77e-30a8aae351f1)

**Comment by develop7** [comment]
Hey @deemp, could you hide the lines in question behind a [lines' group](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines) at least? Seems really straightforward to implement. They should collapse automatically after the end group marker being printed, and do not expand in the log viewer, so that should decrease log pollution.

**Comment by deemp** [comment]
Hi, @develop7! Nice idea. Implemented in 17d19d3d8be918757589635bc1a6830be0b129d2

**Comment by develop7** [comment]
Works like a charm, thanks!
